### ISSUE TITLE
Generated midis are all very similar, all sound like the longest ones from my dataset

### ISSUE THREAD

**Issue by 119b** [tensorflow/magenta#1392]
Hi,

This is probably an issue on my end, and I was wondering if anyone could help. I am very new to all of this. If there is a more appropriate place to ask about this, please let me know.

I built my own dataset and followed instructions for basic_rnn generation. My dataset is small at 66 midis, which very in length from about 3 to 30 seconds.

I did 10,000 training steps with 64, 64, 64. I had it generated 50 midis, each at 128 steps in length with a primer melody of 60.

All of the midis it generated sound very similar, and they sound a lot like the 2 longest midis out of the 66 I supplied it with. Long story short, the output is not meaningful.

I'm not sure where I went wrong. Should I try using lookback or attention instead? Do all the midis I use to train it need to be the same length? Any help would be appreciated

**Comment by adarob** [comment]
It sounds like your dataset is much too small. Since the longest MIDI you included is providing most of the training data, you are likely overfitting on it. You could try adding more dropout and stopping training once you see the evaluation loss increasing, but it would be even better if you added more data.
